src/MagnumPlugins/DdsImporter/DdsImporter.cpp

```diff
@@ -0,0 +1,387 @@
+    /* clear previous data */
+    _imageData2D.clear();
+    _imageData3D.clear();
+
+    _in = new std::istringstream{{data, data.size()}};
+
+    /* read magic number to verify this is a dds file. */
+    char magic[4];
+    _in->read(magic, 4);
+    if (strncmp(magic, "DDS ", 4) != 0) {
+        Error() << "Not a DDS file.";
+    }
+
+    /* read in DDS header */
+    DdsHeader ddsh;
+    _in->read(reinterpret_cast<char*>(&ddsh), sizeof(DdsHeader));
+
+    /* check if image is a 2D or 3D texture */
+    const UnsignedByte dimensions = ((ddsh.caps2 & DdsCaps2::Volume) && (ddsh.depth > 0)) ? 3 : 2;
+
+    /* check if image is a cubemap */
+    const bool isCubemap = (ddsh.caps2 & DdsCaps2::Cubemap);
+
+    bool compressed = false;
+    union {
+        ColorFormat uncompressed;
+        CompressedColorFormat compressed;
+    } colorFormat;
+
+    /* components per pixel */
+    UnsignedInt components = 4;
+
+    /* set the color format */
+    if (ddsh.ddspf.flags & DdsPixelFormat::FourCC) {
+        switch (DdsCompressionTypes(ddsh.ddspf.fourCC)) {
+            case DdsCompressionTypes::DXT1:
+                colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt1;
+                break;
+            case DdsCompressionTypes::DXT3:
+                colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt3;
+                break;
+            case DdsCompressionTypes::DXT5:
+                colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt5;
+                break;
+            default:
+                Error() << "unknown texture compression '" + fourcc(ddsh.ddspf.fourCC) + "'";
+        }
+        compressed = true;
+    } else if (ddsh.ddspf.rgbBitCount == 32 &&
+               ddsh.ddspf.rBitMask == 0x00FF0000 &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x000000FF &&
+               ddsh.ddspf.aBitMask == 0xFF000000) {
+        colorFormat.uncompressed = ColorFormat::BGRA;
+    } else if (ddsh.ddspf.rgbBitCount == 32 &&
+               ddsh.ddspf.rBitMask == 0x000000FF &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x00FF0000 &&
+               ddsh.ddspf.aBitMask == 0xFF000000) {
+        colorFormat.uncompressed = ColorFormat::RGBA;
+    } else if (ddsh.ddspf.rgbBitCount == 24 &&
+               ddsh.ddspf.rBitMask == 0x000000FF &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x00FF0000) {
+        colorFormat.uncompressed = ColorFormat::RGB;
+        components = 3;
+    } else if (ddsh.ddspf.rgbBitCount == 24 &&
+               ddsh.ddspf.rBitMask == 0x00FF0000 &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x000000FF) {
+        colorFormat.uncompressed = ColorFormat::BGR;
+        components = 3;
+    } else if (ddsh.ddspf.rgbBitCount == 8) {
+#ifndef MAGNUM_TARGET_GLES2
```

Comment:
You can indent the preprocessor stuff too, it's not a portability problem as far as I can tell.